builtin/logical/pki/path_roles.go

```diff
@@ -505,35 +513,37 @@ func parseKeyUsages(input string) int {
-	AllowTokenDisplayName bool   `json:"allow_token_displayname" structs:"allow_token_displayname" mapstructure:"allow_token_displayname"`
-	AllowSubdomains       bool   `json:"allow_subdomains" structs:"allow_subdomains" mapstructure:"allow_subdomains"`
-	AllowGlobDomains      bool   `json:"allow_glob_domains" structs:"allow_glob_domains" mapstructure:"allow_glob_domains"`
-	AllowAnyName          bool   `json:"allow_any_name" structs:"allow_any_name" mapstructure:"allow_any_name"`
-	EnforceHostnames      bool   `json:"enforce_hostnames" structs:"enforce_hostnames" mapstructure:"enforce_hostnames"`
-	AllowIPSANs           bool   `json:"allow_ip_sans" structs:"allow_ip_sans" mapstructure:"allow_ip_sans"`
-	ServerFlag            bool   `json:"server_flag" structs:"server_flag" mapstructure:"server_flag"`
-	ClientFlag            bool   `json:"client_flag" structs:"client_flag" mapstructure:"client_flag"`
-	CodeSigningFlag       bool   `json:"code_signing_flag" structs:"code_signing_flag" mapstructure:"code_signing_flag"`
-	EmailProtectionFlag   bool   `json:"email_protection_flag" structs:"email_protection_flag" mapstructure:"email_protection_flag"`
-	UseCSRCommonName      bool   `json:"use_csr_common_name" structs:"use_csr_common_name" mapstructure:"use_csr_common_name"`
-	UseCSRSANs            bool   `json:"use_csr_sans" structs:"use_csr_sans" mapstructure:"use_csr_sans"`
-	KeyType               string `json:"key_type" structs:"key_type" mapstructure:"key_type"`
-	KeyBits               int    `json:"key_bits" structs:"key_bits" mapstructure:"key_bits"`
-	MaxPathLength         *int   `json:",omitempty" structs:"max_path_length,omitempty" mapstructure:"max_path_length"`
-	KeyUsage              string `json:"key_usage" structs:"key_usage" mapstructure:"key_usage"`
-	OU                    string `json:"ou" structs:"ou" mapstructure:"ou"`
-	Organization          string `json:"organization" structs:"organization" mapstructure:"organization"`
-	GenerateLease         *bool  `json:"generate_lease,omitempty" structs:"generate_lease,omitempty"`
-	NoStore               bool   `json:"no_store" structs:"no_store" mapstructure:"no_store"`
+	LeaseMax              string   `json:"lease_max" structs:"lease_max" mapstructure:"lease_max"`
+	Lease                 string   `json:"lease" structs:"lease" mapstructure:"lease"`
+	MaxTTL                string   `json:"max_ttl" structs:"max_ttl" mapstructure:"max_ttl"`
+	TTL                   string   `json:"ttl" structs:"ttl" mapstructure:"ttl"`
+	AllowLocalhost        bool     `json:"allow_localhost" structs:"allow_localhost" mapstructure:"allow_localhost"`
+	AllowedBaseDomain     string   `json:"allowed_base_domain" structs:"allowed_base_domain" mapstructure:"allowed_base_domain"`
+	AllowedDomainsOld     string   `json:"allowed_domains,omit_empty" structs:"allowed_domains_old" mapstructure:"allowed_domains_old"`
```

Comment:
The AllowedDomainsOld and KeyUsageOld is retaining the old JSON tags and the new one has `_list` appended, which makes sense. Do we need the structs and mapstructure tags for the old fields?

Reply:
I did some funny business with structs and mapstructure tags.  This package uses structs and mapstructure for various translations.  I wanted to use the old field names for the new fields so even though I kept the old json tags the same, I changed the structs and mapstructure tags so I can delete them later.  It may be more explicit to do the translation directly in the responses but I was able to get the responses I wanted just by doing this tag business.  I'm curious your thoughts on if it makes sense to do it this way or should I be more explicit and do the translation before returning the response and make these tags more clear.

Reply:
Yes, PKI's origins predate my current dislike of structs and mapstructure :-)  Future refactoring work will hopefully get rid of those; if you feel like spending the time now, it may make things cleaner.